sphinx/builders/linkcheck.py

```diff
@@ -706,15 +713,37 @@ def contains_anchor(response: Response, anchor: str) -> bool:
 class AnchorCheckParser(HTMLParser):
     """Specialised HTML parser that looks for a specific anchor."""
 
-    def __init__(self, search_anchor: str) -> None:
+    def __init__(self, search_anchor: str, original_encoded_anchor: str = '') -> None:
```

Comment:
A function call that contains the same value encoded in two different ways is, to me, something of an undesirable code smell.  It's probably not the first item to focus on, but it would be nice if this initializer could be updated to accept only one form of the anchor (I'd probably recommend the original value received before any encoding-related operations have been attempted).